src/lib/assetBridger/l1l3Bridger.ts

```diff
@@ -0,0 +1,1166 @@
+        `Unknown l1 network chain id: ${l2Network.partnerChainID}`
+      )
+    }
+
+    this.l1Network = l1Network
+    this.l2Network = l2Network
+    this.l3Network = l3Network
+  }
+
+  /**
+   * Check the signer/provider matches the l1Network, throws if not
+   * @param sop
+   */
+  protected async _checkL1Network(sop: SignerOrProvider): Promise<void> {
+    await SignerProviderUtils.checkNetworkMatches(sop, this.l1Network.chainID)
+  }
+
+  /**
+   * Check the signer/provider matches the l2Network, throws if not
+   * @param sop
+   */
+  protected async _checkL2Network(sop: SignerOrProvider): Promise<void> {
+    await SignerProviderUtils.checkNetworkMatches(sop, this.l2Network.chainID)
+  }
+
+  /**
+   * Check the signer/provider matches the l2Network, throws if not
+   * @param sop
+   */
+  protected async _checkL3Network(sop: SignerOrProvider): Promise<void> {
+    await SignerProviderUtils.checkNetworkMatches(sop, this.l3Network.chainID)
+  }
+
+  protected _percentIncrease(num: BigNumber, increase: BigNumber): BigNumber {
+    return num.add(num.mul(increase).div(100))
+  }
+}
+
+/**
+ * Base functionality for bridging ERC20 tokens from L1 to L3.
+ */
+class BaseErc20L1L3Bridger extends BaseL1L3Bridger {
+  public readonly teleporterAddresses: TeleporterAddresses
+
+  // todo: tune these
+  public readonly defaultRetryableGasParams: ManualRetryableGasParams = {
+    l2ForwarderFactoryGasLimit: BigNumber.from(1_000_000),
+    l1l2TokenBridgeGasLimit: BigNumber.from(1_000_000),
+    l2l3TokenBridgeGasLimit: BigNumber.from(1_000_000),
+    l1l2TokenBridgeRetryableSize: BigNumber.from(1000),
+    l2l3TokenBridgeRetryableSize: BigNumber.from(1000),
+  } as const
+
+  public constructor(public readonly l3Network: L2Network) {
+    super(l3Network)
+
+    if (!this.l2Network.teleporterAddresses) {
+      throw new ArbSdkError(
+        `L2 network ${this.l2Network.name} does not have teleporter contracts`
+      )
+    }
+
+    this.teleporterAddresses = this.l2Network.teleporterAddresses
+  }
+
+  /**
+   * Get the corresponding L2 token address for the provided L1 token
+   */
+  public async getL2ERC20Address(
+    erc20L1Address: string,
+    l1Provider: Provider
+  ): Promise<string> {
+    await this._checkL1Network(l1Provider)
+    return this._getChildErc20Address(
+      erc20L1Address,
+      l1Provider,
+      this.l2Network
+    )
+  }
+
+  /**
+   * Get the corresponding L3 token address for the provided L1 token
+   */
+  public async getL3ERC20Address(
+    erc20L1Address: string,
+    l1Provider: Provider,
+    l2Provider: Provider
+  ): Promise<string> {
+    await this._checkL2Network(l2Provider)
+    return this._getChildErc20Address(
+      await this.getL2ERC20Address(erc20L1Address, l1Provider),
+      l2Provider,
+      this.l3Network
+    )
+  }
+
+  /**
+   * Given an erc20 address on a parent network, get the corresponding erc20 address on a child network
+   */
+  private _getChildErc20Address(
```

Comment:
code in erc20Bridger.getL2ERC20Address could be reused (not necessary tho)

Reply:
agree that's nicer